app/experimenter/experiments/forms.py

```diff
@@ -314,20 +315,79 @@ class ExperimentVariantsAddonForm(ChangeLogMixin, forms.ModelForm):
         widget=forms.Textarea(attrs={"class": "form-control", "rows": 10}),
     )
 
+    ALL_KEY = "__all__"
+    ALL_LABEL = "ALL"
+
+    # The reason we do locales and countries with a custom non-model
+    # MultipleChoiceField is to allow in a value that can never
+    # be found in the Locale.objects.all QuerySet.
+    # We would not be able to execute "hacks" in the `clean_locales`
+    # and `clean_countries` method hooks since it wouldn't be
+    # called if a 'code' is passed in. Django's model forms would
+    # never allow it.
+
+    # By making it a non-model form field, we "sneak" in the custom
+    # "__all__" choice and use the form's `clean` method hook to
+    # remove it as a choice and potentially transfer this presence
+    # to another field (`all_locales` & `all_countries`).
+
+    locales = forms.MultipleChoiceField(
+        label="Locale(s)",
+        required=False,
+        help_text="Applicable only if you don't select All",
+    )
+    countries = forms.MultipleChoiceField(
+        label="Geographic Region(s)",
+        required=False,
+        help_text="Applicable only if you don't select All",
+    )
+    # See https://developer.snapappointments.com/bootstrap-select/examples/
+    # for more options that relate to the initial rendering of the HTML
+    # as a way to customize how it works.
+    locales.widget.attrs.update({"data-live-search": "true"})
+    countries.widget.attrs.update({"data-live-search": "true"})
+
     class Meta:
         model = Experiment
         fields = [
             "population_percent",
             "firefox_version",
             "firefox_channel",
             "client_matching",
+            "locales",
+            "all_locales",
+            "countries",
+            "all_countries",
         ]
 
     def __init__(self, *args, **kwargs):
         data = kwargs.pop("data", None)
         instance = kwargs.pop("instance", None)
+        if instance:
```

Comment:
I'd like to see this logic moved out of the form and into a subclass of MultipleChoiceField if possible.

Reply:
I wish we could move all of this logic out of the form because right now those form fields are dependent on this logic also being in the form to work properly, but it would require passing the instance into the form fields so I think we can leave it like this and refactor it in the future if necessary.